api_tutorials/bait_cnn/bait_performance_cnn.py

```diff
@@ -0,0 +1,605 @@
+                      'V': [0.333, 0.333, 0, 0.334],
+                      'H': [0.333, 0.333, 0.334, 0],
+                      'D': [0.333, 0, 0.333, 0.334],
+                      'X': [0.25, 0.25, 0.25, 0.25],
+                      'N': [0.25, 0.25, 0.25, 0.25]}
+
+    count = 0
+    np.random.seed(0)
+    while count < args.samples:
+        contig_key, start, end, coverage, gc = sample_from_bed(baits_and_coverages)
+        mid = (start + end) // 2
+        contig = record_dict[contig_key]
+
+        record = contig[mid - idx_offset: mid + idx_offset]
+
+        train_data.coverage[count, 0] = coverage
+        train_data.gc[count] = gc
+
+        for i, b in enumerate(record.seq):
+            B = b.upper()
+            if B in args.inputs.keys():
+                train_data.baitdata[count, i, args.inputs[B]] = 1.0
+            elif B in amiguity_codes.keys():
+                train_data.baitdata[count, i, 0:4] = amiguity_codes[B]
+            else:
+                print('Error! Unknown code:', b)
+                return
+
+            ref_pos = i + mid - idx_offset
+            if start <= ref_pos <= end:
+                train_data.baitdata[count, i, 4] = 1
+
+        count += 1
+
+    print('Train data shape: %s' % (train_data.baitdata.shape,))
+    print('Coverage data shape: %s' % (train_data.coverage.shape,))
+
+    # should be in the form ( 5xsize x samples tensor, 1x samples tensor)
+    return train_data
+
+
+def bed_file_to_coverage(bed_file):
+    bed_with_cov_and_gc = {}
+
+    total_reads = 0L  # type: long
+    total_baits = 0L  # type: long
+
+    with open(bed_file) as f:
+        for line in f:
+            parts = line.split()
+            contig = parts[0]
+            lower = int(parts[1])
+            upper = int(parts[2])
+            gc = float(parts[3])
+            reads = int(parts[4])
+            total_reads += reads
+            total_baits += 1
+
+            if contig not in bed_with_cov_and_gc.keys():
+                bed_with_cov_and_gc[contig] = ([], [], [], [],)
+
+            bed_with_cov_and_gc[contig][0].append(lower)
+            bed_with_cov_and_gc[contig][1].append(upper)
+            bed_with_cov_and_gc[contig][2].append(reads)
+            bed_with_cov_and_gc[contig][3].append(gc)
+
+    reads_per_bait = float(total_reads) / total_baits
+    print("Reads per bait: %s" % reads_per_bait)
+    for contig in bed_with_cov_and_gc.keys():
+        bed_with_cov_and_gc[contig] = (
+            np.array(bed_with_cov_and_gc[contig][0]),
+            np.array(bed_with_cov_and_gc[contig][1]),
+            np.array([x / reads_per_bait for x in bed_with_cov_and_gc[contig][2]]),
+            bed_with_cov_and_gc[contig][3],)
+        print('key is: %s len %s' % (contig, len(bed_with_cov_and_gc[contig][0])))
+
+    return bed_with_cov_and_gc
+
+
+def in_bed_file(bed_dict, contig, pos):
+    lows = bed_dict[contig][0]
+    ups = bed_dict[contig][1]
+
+    return np.any((lows <= pos) & (pos <= ups))
```

Comment:
Again my bug which got fixed in vqsr but never cherrypicked back here.  I believe this sould be:
`return np.any((lows <= pos) & (pos < ups))`